src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -63,7 +63,68 @@ public static int IndexOf(ref byte searchSpace, int searchSpaceLength, ref byte
             // Based on http://0x80.pl/articles/simd-strfind.html#algorithm-1-generic-simd "Algorithm 1: Generic SIMD" by Wojciech Mula
             // Some details about the implementation can also be found in https://github.com/dotnet/runtime/pull/63285
         SEARCH_TWO_BYTES:
-            if (Vector256.IsHardwareAccelerated && searchSpaceMinusValueTailLength - Vector256<byte>.Count >= 0)
+            if (Vector512.IsHardwareAccelerated && searchSpaceMinusValueTailLength - Vector512<byte>.Count >= 0)
```

Comment:
Given the current handling we have for first gen AVX-512 hardware in the JIT, we probably want to have an internal helper and make this check something like:
```suggestion
if (Vector512.IsHardwareAccelerated && !X86Base.Vector512Throttling && searchSpaceMinusValueTailLength - Vector512<byte>.Count >= 0)
```

The `X86Base.Vector512Throttling` check can just be an `internal static bool Vector512Throttling` intrinsic that returns the underlying `JitFlags`.

Reply:
Does this make sense more as a flag/variable similar to 'IsSupported' or as an function to be inlined. I'm leaning more the former

Reply:
Would I need to hide a hypothetical `X86Base.Vector512Throttling` check inside a `X86Base.IsSupported`or `CompatDependsOn `?

Reply:
I don't think so since its just returning a `bool` and and `X86Base` is considered part of the baseline ISA set, so isn't tracked.

Reply:
However, we could put it on `Vector512` instead just to be safe. That would make it `Vector512.IsHardwareAccelerated && !Vector512.IsThrottling`, which seems "clear enough" for an internal only intrinsic.

Reply:
Yeah - I think I prefer this too.

Reply:
@tannergooding I'm having some trouble adding an internal property in Vector512. Outside of in Vector512.cs(https://github.com/dotnet/runtime/pull/86655/commits/49aa3168ef18de84b5d37ce47bd9c0400a3aa360#diff-97c3213a2a53c89fc5c8fd05ea68767d5b37e4fc5907825e814c150a55e3870dR55-R59), where else in library do I have to update? I added in JIT I think but I'm getting an 'UnSupported' exception
![image](https://github.com/dotnet/runtime/assets/17969209/20294843-4f8e-4c5a-ad97-b1443be252ba)

Reply:
You should just need to update it there and ensure it's handled in the JIT alongside the existing `IsHardwareAccelerated`/`IsSupported` handling.

I'd probably not make it recursive since its internal only. That avoids needing to ensure Mono handles it as well.

Reply:
As in the method is currently defined as recursively calling itself:
```
internal static bool Vector512Throttling
{
    [Intrinsic]
    get => Vector512Throttling;
}
```

This is primarily important so that indirect invocation still works as expected and requires special JIT handling. That special JIT handling works implicitly for a `IsHardwareAccelerated` and `IsSupported`. New members, like `get_IsThrottled`, would need additional handling. Given that the API is internal, this support for indirect invocationisn't important and we can instead have it be `get => false;` which means it can only return `true` if called directly and if the runtime intrinsifies it and it means that Mono doesn't need to add explicit support for the new recursive member.

----------------------

An alternative is that we don't define this internal property at all. Instead, we'd have it surface indirectly via `IsHardwareAccelerated` itself, which allows external users to depend on it. That is, I think we have two viable paths here, each with their own tradeoffs. I don't have a strong preference for which we do in this PR. We probably want to discuss and determine what we think the best option before shipping is, however.

The first option is to have this internal `IsThrottled`/`IsThrottling` property and have the BCL use it as `Vector512.IsHardwareAccelerated && !Vector512.IsThrottled`. This allows the BCL to easily depend on it. However, it means users may end up needing to have their own equivalent to our internal `IsThrottled` property, replicating the checks we have internally.

The second option is to instead surface this directly through `Vector512.IsHardwareAccelerated`. That is, the JIT only returns `true` for `IsHardwareAccelerated` if `compSupportsHWIntrinsic(InstructionSet_AVX512F) && !jitFlags.IsSet(JitFlags::JIT_FLAG_VECTOR512_THROTTLING)`. The actual implementation would still use acceleration, much as `Vector256` does on hardware with AVX but without AVX2. The downside is that users on first generation hardware would have to explicitly check `Avx512F.IsSupported` if they wanted to light-up.

Its the same general support/consideration, just differing slightly in how its exposed and whether/how its visible to 3rd party consumers of the API.

Reply:
I imagine we would eventually need to expose a 'IsThrottled' property anyway even for AVx512F for users who want to skip using AVX512 features when throttling is an issue. Considering that, I think option A(Exposing property) makes sense(maybe eventually making it public?).

Reply:
Either option gives that ability. It'd effectively be...

Option A:
* `Vector512.IsHardwareAccelerated` -- I don't care about throttling
* `Vector512.IsHardwareAccelerated && !Vector512.IsThrottled` -- I care about throttling

Option B:
* `Vector512.IsHardwareAccelerated` -- I care about throttling
* `Avx512F.IsSupported` -- I don't care about throttling

The main downside of `Option A` is that gives users "one more thing" to think about when presumably the common case is they want things to work just like `Vector128`/`Vector256`. It's also a bit quirky in that its exposing an xarch specific concern as part of the general API surface.

The main downside of `Option B` ends up being that users who are doing HPC or large workloads need to tie that functionality directly to an `Avx512F.IsSupported` check. 

It's something we should discuss a bit more in our next sync meeting. What we do for this PR, particularly while its internal, doesn't matter too much and we can always tweak it later.

Reply:
Ah! Got it. Yeah- that makes sense.

Assuming this is going to be revisited later, I'm going to keep it to Option B. That way I'm not adding anything new to API surface which needs to be taken out. Adding something later seems better to me that taking it out. I'm okay with either though

Reply:
Works for me! Thanks for the discussion!